contracts/drafts/ERC1820.sol

```diff
@@ -0,0 +1,25 @@
+pragma solidity ^0.5.2;
+
+import "./IERC1820.sol";
+
+/**
+ * Inherit from this contract and call _registerInterfaceForAddress to allow for contracts to be registered in the
+ * ERC1820 registry.
+ */
+contract ERC1820 is IERC1820 {
+    bytes32 constant private ERC1820_ACCEPT_MAGIC = keccak256(abi.encodePacked("ERC1820_ACCEPT_MAGIC"));
+
+    mapping(bytes32 => mapping(address => bool)) private _supportedInterfaces;
+
+    function canImplementInterfaceForAddress(bytes32 interfaceHash, address account) external view returns (bytes32) {
+        return _implementsInterfaceForAddress(interfaceHash, account) ? ERC1820_ACCEPT_MAGIC : bytes32(0x00);
+    }
+
+    function _implementsInterfaceForAddress(bytes32 interfaceHash, address account) internal view returns (bool) {
```

Comment:
I'm hesitant to include this as part of the `internal` API. Do we think it will be needed? I'd make it `private` or remove it altogether.

Reply:
I added it since going through the `public` function requires knowledge of the magic number, but on second thought I can't think of many scenarios where such a function would be needed. I'm all for simplifying this.